verilog/preprocessor/verilog_preprocess_test.cc

```diff
@@ -264,5 +269,260 @@ TEST(VerilogPreprocessTest, DefaultPreprocessorKeepsDefineInStream) {
   EXPECT_EQ(count_defines, 2);
 }
 
+struct BranchFailTest {
+  const char* input;
+  int offset;
+  const char* expected_error;
+};
+TEST(VerilogPreprocessTest, IncompleteOrUnbalancedIfdef) {
+  const BranchFailTest test_cases[] = {
+      {"`endif", 0, "Unmatched `endif"},
+      {"`else", 0, "Unmatched `else"},
+      {"`ifdef", 6, "unexpected EOF where expecting macro name"},
+      {"`ifdef FOO\n`endif\n`endif", 18, "Unmatched `endif"},
+      {"`ifdef FOO\n`endif\n`else", 18, "Unmatched `else"},
+      {"`ifdef FOO\n`endif\n`elsif BAR", 18, "Unmatched `elsif"},
+      {"`ifdef FOO\n`else\n`else", 17, "Duplicate `else"},
+      {"`ifdef FOO\n`else\n`elsif BAR", 17, "`elsif after `else"},
+      {"`ifdef FOO\n`ifdef BAR`else\n`else", 27, "Duplicate `else"},
+      {"`ifdef FOO\n`else\n`ifdef BAR\n`endif", 11, "Branch started here"},
+      {"`ifdef FOO\n`elsif BAR\n", 11, "Branch started here"},
+      {"`ifdef FOO\n`elsif BAR\n`else\n", 22, "Branch started here"},
+  };
+  for (const BranchFailTest& test : test_cases) {
+    PreprocessorTester tester(
+        test.input, VerilogPreprocess::Config({.filter_branches = true}));
+
+    EXPECT_FALSE(tester.Status().ok());
```

Comment:
Do we have `EXPECT_OK`?

Reply:
No, we usually #define it ourself. And here, we'd have to define `EXPECT_NOT_OK` ... so the current form is more readable.